code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,703 @@
+      operationId: deleteArtifact
+      parameters:
+        - name: artifactId
+          in: path
+          description: Identificator of the artifact to be deleted provided by the OP NBI once the uploading was successful
+          required: true
+          schema:
+            $ref: "#/components/schemas/ArtifactId"
+
+      responses:
+        '204':
+          description: Artifact deleted
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '409':
+          description: Conflict
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorInfo'
+              example:
+                status: 409
+                code: CONFLICT
+                message: "Artifact with a running application cannot be deleted"
+        '500':
+          $ref: '#/components/responses/500'
+        '503':
+          $ref: '#/components/responses/503'
+
+  /edgeCloudPlatform:
+    get:
+      tags: 
+        - Mobile Edge Cloud Platform
+      summary: Retrieve a list of the operator’s MECs and their status
+      description: List of the operator’s MECs and their status, ordering the results by location and filtering by status (active/inactive/unknown)
+      operationId: getMecPlatforms
+      parameters:
+        - name: regionId
+          description: Human readable name of the geografical zone of the MEC. Defined by the OP.
+          in: query
+          required: false
+          schema: 
+              $ref: '#/components/schemas/RegionId'
+        - name: status
+          in: query
+          required: false
+          schema: 
+              $ref: '#/components/schemas/Status'
+      responses:
+        '200':
+          description: Available MEC Platforms 
+          content:
+            application/json:
+              schema:
+                type: array
+                items:                  
+                  $ref: '#/components/schemas/MecDetails'
+                minItems: 1
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '500':
+          $ref: '#/components/responses/500'
+        '503':
+          $ref: '#/components/responses/503'
+
+components:
+  securitySchemes:
+    oAuth2ClientCredentials:
+      type: oauth2
+      flows: 
+        clientCredentials: 
+          tokenUrl: '/oauth2/token'
+          scopes:
+            nbi-mgmt: Access to the APIs
+  
+  schemas:
+    AppId:
+      type: string
+      format: uuid
+      description: A globally unique identifier assosiated with the artifact. Originating OP generates this identifier when artifact is submitted over NBI.
+       
+    AppProviderId:
```

Comment:
Do we really need an explicit appProviderId? As the API is to be authorized by oAuth2.0, the oAuth credentials should be able to determine the identity of the app provider.

Reply:
Agree with that. Parameter removed.

[Commit](https://github.com/camaraproject/EdgeCloud/pull/154/commits/3ffe30c6c302393e4c9a0850bc2879ad5e8bf6e5)

Reply:
My preference would be to leave user/roles/permissions management APIs out of this spec, as every operator will have their own approach to this. The intent here is to make it possible for such a management system to be able to manage separate permissions for groups of objects (i.e. a Project that consists of apps, app instances, etc). This is a pretty standard approach.

Reply:
I agree with the roles or permission management part which can be out of this spec. Also, i would suggest to keep the appProviderId for future consideration and keep the API bit simple in first version unless there are very compelling reason to keep it.

Reply:
Agree @gunjald , for this fisrt MVP it is assumed that the authentication has already been done outside the API scope. Anyhow we find this usefull and could be integrated in future versions.